Tools/TestWebKitAPI/Tests/WebKitCocoa/IDBCheckpointWAL.mm

```diff
@@ -86,29 +86,29 @@ long long fileSizeAtPath(NSString *path)
 
     [configuration.get().websiteDataStore _terminateNetworkProcess];
 
-    auto webView = adoptNS([[WKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:configuration.get()]);
-    NSURLRequest *request = [NSURLRequest requestWithURL:[[NSBundle mainBundle] URLForResource:@"IDBCheckpointWAL" withExtension:@"html" subdirectory:@"TestWebKitAPI.resources"]];
-    [webView loadRequest:request];
+    RetainPtr webView = adoptNS([[WKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:configuration.get()]);
+    RetainPtr request = [NSURLRequest requestWithURL:[[NSBundle mainBundle] URLForResource:@"IDBCheckpointWAL" withExtension:@"html" subdirectory:@"TestWebKitAPI.resources"]];
+    [webView loadRequest:request.get()];
     TestWebKitAPI::Util::run([handler receivedScriptMessagePointer]);
     EXPECT_STREQ([[[handler lastScriptMessage] body] UTF8String], "Success");
 
-    NSURL *originURL = [NSURL URLWithString:@"file://"];
-    __block NSString *indexedDBDirectoryString = nil;
-    [configuration.get().websiteDataStore _originDirectoryForTesting:originURL topOrigin:originURL type:WKWebsiteDataTypeIndexedDBDatabases completionHandler:^(NSString *result) {
+    RetainPtr originURL = [NSURL URLWithString:@"file://"];
+    __block RetainPtr<NSString> indexedDBDirectoryString;
```

Comment:
These were particularly unsafe since we run some async operation and then expected the string to stay alive later on.